packages/reactive_ble_mobile/ios/Classes/ReactiveBle/Central.swift

```diff
@@ -226,8 +226,11 @@ final class Central {
 
         guard characteristic.properties.contains(.read)
         else { throw Failure.notReadable(qualifiedCharacteristic) }
-
-        characteristic.service.peripheral.readValue(for: characteristic)
+        
+        guard let value =  characteristic.service?.peripheral?.readValue(for: characteristic) else{
+            throw Failure.characteristicNotFound(qualifiedCharacteristic)
+        }
+        return value
```

Comment:
You're returning `void` here.

1. Remove `return`.
2. Do `guard let peripheral = characteristic.service?.peripheral` (up to naming).
3. Do `peripheral.readValue(...)` after the guard.

And, please, fix formatting for all guards you introduced in this p/r so that they look like similar to the following examples.

```swift
guard let x = x // A single binding
else { return_or_throw(); } // A short (single statement) else-branch

guard
    let x = x,
    let y = y // Multiple bindings
else { return_or_throw(); }  // A short (single statement) else-branch

guard let x = x
else {  // A long (multiple statements) else-branch
    do_something();
    return_or_throw();
}
```

Don't forget about spaces (as in `else {`, not `else{`).

This is the style used in the (original) Swift code-base, even if Xcode and/or some third-party guides suggest different formatting.

Reply:
fixed!